merged inbox.lv config with inbox.lt and added inbox.eu, mail.ee, inbox.la

jmalinens opened the pull request:
These domains have the same configuration and owned by the same e-mail provider

jmalinens commented:
We already have configs on our side but ispdb data is outdated (does not include IMAP):

http://autoconfig.inbox.lv/mail/config-v1.1.xml
http://autoconfig.inbox.lt/mail/config-v1.1.xml
http://autoconfig.inbox.eu/mail/config-v1.1.xml
http://autoconfig.inbox.la/mail/config-v1.1.xml
http://autoconfig.mail.ee/mail/config-v1.1.xml

benbucksch commented on ispdb/inbox.lv:
```diff
@@ -2,25 +2,49 @@
+      <username>%EMAILADDRESS%</username>
       <authentication>password-cleartext</authentication>
     </outgoingServer>
+    <documentation url="https://help.%EMAILDOMAIN%/?keyword=email_clients_settings">
```
You cannot use placeholders in this URL field.

benbucksch commented on ispdb/inbox.lv:
```diff
@@ -2,25 +2,49 @@
+       <descr lang="lt">Pašto programų nustatymai (IMAP, POP3, SMTP)</descr>
+       <descr lang="ru">Настройки для почтовых программ (IMAP, POP3,SMTP и др.)</descr>
+    </documentation>
   </emailProvider>
```
Please add:
```
<enable visiturl="https://help.inbox.eu/help/question/443/37">
  <instruction>Enable external POP3/IMAP and SMTP access in your account settings</instruction>
  <instruction lang="lv">...</instruction>
</enable>
```
and add translations for the main languages of this ISP.

benbucksch replied:
Same for the other 2 config files.

benbucksch replied:
Note that no placeholders are allowed in the URL. However, it would be OK to use inbox.eu, for example, for all the ISPs. Don't worry too much about the URL, the page is not currently used, but we may add a feature to show this page to the user. At the least, it captures the information for us here in the file, in a structured form.

benbucksch commented on ispdb/inbox.lv:
```diff
@@ -2,25 +2,49 @@
-      <hostname>mail.inbox.lv</hostname>
+      <hostname>mail.%EMAILDOMAIN%</hostname>
       <port>587</port>
       <socketType>STARTTLS</socketType>
```
Please use SSL and 465 here. This matches the incoming servers.
Same for the other configs below.

benbucksch commented on ispdb/tvnet.lv:
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<clientConfig version="1.1">
+  <emailProvider id="tvnet.lv">
```
Does this domain have significant usage, so that it merits to be added here? We only add large ISPs.

jmalinens replied:
Thank You for the review.
I will make changes Monday accordingly!
Tvnet.lv active usage is in tens of thousands users but does not exceed 100,000 mark.
I can remove it if it is not enough.
I work for this ISP and am one of the developers there.
I will test accounts on all domains to check if I missed something.
Most domains work the same with these differences:

* mail.ee has IMAP/SMTP as mail.ee instead of mail.mail.ee, other domains have mail.domain.tld
* tvnet.lv users use mail.inbox.lv SMTP/IMAP directly instead of mail.tvnet.lv

benbucksch commented:
> We already have configs on our side

@jmalinens: Do you work for that ISP? In which function? Either way: Thanks for your submission!

benbucksch commented:
If you have corresponding accounts, could you please test this locally? Search for "harddisk" on https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration .

jmalinens commented:
@benbucksch please review my changes.

I have:
* removed tvnet.lv config
* changed SMTP to 465/SSL
* added <enable> tag
* hardcoded documentation link to help.inbox.lv instead of help.%EMAILDOMAIN%

Unfortunately i could not check the configs with harddisk method as it looks like it falls back to http://autoconfig.%EMAILDOMAIN%/mail/config-v1.1.xml config.

I work on linux and tried to add inbox.lv.xml in ~/.thunderbird/isp/inbox.lv.xml and restarted thunderbird but it does not seem to settings defined there.

benbucksch commented:
@jmalinens , thanks for the changes!

> I work on linux and tried to add inbox.lv.xml in ~/.thunderbird/isp/inbox.lv.xml and restarted 

The file needs to be in the Thunderbird installation directory, not the profile folder in your home dir.

That is, if you download and unpack thunderbird tar.gz in `/foo/`, and thunderbird executable is in `/foo/thunderbird/thunderbird`, then the file needs to be `/foo/thunderbird/isp/inbox.lv.xml`.

> it looks like it falls back to http://autoconfig.%EMAILDOMAIN%/mail/config-v1.1.xml config

If that happens, then either the filename is wrong, or the file has a syntax error. In the latter case, we cannot deploy it, so it's important that you see it working.

Also, I am not sure whether we do variable replacement in the display name. Could you please verify that this works?

If you verified that it works as is, it can be merged.

jmalinens commented:
@benbucksch  I confirm it works via harddisk method with Message:
Configuration found on Thunderbrid installation

The correct path was: /usr/lib/thunderbird/isp

benbucksch commented:
Perfect, thank you!

Do you also see the expected display name?

benbucksch commented:
> Do you also see the expected display name?

It should work in theory, see:
https://searchfox.org/comm-central/source/mail/components/accountcreation/content/accountConfig.js#425
```
account.displayName = _replaceVariable(account.displayName, otherVariables);
```
but it would be good to verify before deploying.

Merged to master.

jmalinens commented:
@benbucksch I don't see where displayName is used in Thunderbird when setting up existing account.
Only full e-mail address, IMAP/SMTP hostnames and ports are used.

Thank You for merging changes!

benbucksch commented:
> Only full e-mail address, IMAP/SMTP hostnames and ports are used.

Alright, no problem. Thanks for checking!
